applications/HDF5Application/tests/test_hdf5_model_part_io.py

```diff
@@ -179,21 +201,56 @@ def test_HDF5NodalSolutionStepDataIO(self):
+        with ControlledExecutionScope(os.path.dirname(os.path.realpath(__file__))):
+            write_model_part = ModelPart("write")
+            self._initialize_model_part(write_model_part)
+            hdf5_file = self._get_file()
+            hdf5_model_part_io = self._get_model_part_io(hdf5_file)
+            hdf5_nodal_data_value_io = self._get_nodal_data_value_io(hdf5_file)
+            hdf5_model_part_io.WriteModelPart(write_model_part)
+            hdf5_nodal_data_value_io.WriteNodalResults(write_model_part.Nodes)
+            read_model_part = ModelPart("read")
+            hdf5_model_part_io.ReadModelPart(read_model_part)
+            hdf5_nodal_data_value_io.ReadNodalResults(read_model_part.Nodes, read_model_part.GetCommunicator())
+
+            assert_variables_list = [DISPLACEMENT_X, DISPLACEMENT_Y, DISPLACEMENT_Z,
+                                     VELOCITY_X, VELOCITY_Y, VELOCITY_Z,
+                                     ACCELERATION_X, ACCELERATION_Y, ACCELERATION_Z,
+                                     PRESSURE, VISCOSITY, DENSITY, ACTIVATION_LEVEL]
             # Check data.
             for read_node, write_node in zip(read_model_part.Nodes, write_model_part.Nodes):
-                self.assertEqual(read_node.GetSolutionStepValue(DISPLACEMENT_X), write_node.GetSolutionStepValue(DISPLACEMENT_X))
-                self.assertEqual(read_node.GetSolutionStepValue(DISPLACEMENT_Y), write_node.GetSolutionStepValue(DISPLACEMENT_Y))
-                self.assertEqual(read_node.GetSolutionStepValue(DISPLACEMENT_Z), write_node.GetSolutionStepValue(DISPLACEMENT_Z))
-                self.assertEqual(read_node.GetSolutionStepValue(VELOCITY_X), write_node.GetSolutionStepValue(VELOCITY_X))
-                self.assertEqual(read_node.GetSolutionStepValue(VELOCITY_Y), write_node.GetSolutionStepValue(VELOCITY_Y))
-                self.assertEqual(read_node.GetSolutionStepValue(VELOCITY_Z), write_node.GetSolutionStepValue(VELOCITY_Z))
-                self.assertEqual(read_node.GetSolutionStepValue(ACCELERATION_X), write_node.GetSolutionStepValue(ACCELERATION_X))
-                self.assertEqual(read_node.GetSolutionStepValue(ACCELERATION_Y), write_node.GetSolutionStepValue(ACCELERATION_Y))
-                self.assertEqual(read_node.GetSolutionStepValue(ACCELERATION_Z), write_node.GetSolutionStepValue(ACCELERATION_Z))
-                self.assertEqual(read_node.GetSolutionStepValue(PRESSURE), write_node.GetSolutionStepValue(PRESSURE))
-                self.assertEqual(read_node.GetSolutionStepValue(VISCOSITY), write_node.GetSolutionStepValue(VISCOSITY))
-                self.assertEqual(read_node.GetSolutionStepValue(DENSITY), write_node.GetSolutionStepValue(DENSITY))
-                self.assertEqual(read_node.GetSolutionStepValue(ACTIVATION_LEVEL), write_node.GetSolutionStepValue(ACTIVATION_LEVEL))
+                for var in assert_variables_list:
+                    self.assertEqual(read_node.GetValue(var), write_node.GetValue(var))
```

Comment:
In `_initialize_model_part` it looks like we forgot to assign nodal data values.

Reply:
I'm sorry, my mistake... Now it is added... :)